### PR TITLE
Add fn support_group_by_expr to Dialect trait

### PR DESCRIPTION
**Pull request by jdye64**
This closes: #895

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -975,7 +975,7 @@ impl<'a> Parser<'a> {
     /// parse a group by expr. a group by expr can be one of group sets, roll up, cube, or simple
     /// expr.
     fn parse_group_by_expr(&mut self) -> Result<Expr, ParserError> {
-        if dialect_of!(self is PostgreSqlDialect | DuckDbDialect | GenericDialect) {
+        if self.dialect.supports_filter_during_aggregation() {
```
Comment (andygrove) [inline src/parser.rs:978]: I think this should be:

```suggestion
        if self.dialect.supports_group_by_expr() {
```

Comment (jdye64) [reply src/parser.rs]: Thanks for catching that Andy. Seems my autocomplete got a little excited there.

**Review by andygrove — APPROVED** [review]
LGTM

**Comment by andygrove** [comment]
cc @Jimexist @alamb

**Review by alamb — APPROVED** [review]
Thank you @jdye64

**Pull request merged**